robot/web/stream/server.go

```diff
@@ -336,6 +336,8 @@ func (server *Server) GetStreamOptions(
 	ctx context.Context,
 	req *streampb.GetStreamOptionsRequest,
 ) (*streampb.GetStreamOptionsResponse, error) {
+	server.mu.RLock()
```

Comment:
Why do we need to add this read lock?

Reply:
The mutex is protecting these struct members:
```
	nameToStreamState       map[string]*state.StreamState
	activePeerStreams       map[*webrtc.PeerConnection]map[string]*peerState
	activeBackgroundWorkers sync.WaitGroup
	isAlive                 bool

```

I don't see them being accessed so I don't understand why this mutex is being taken

Reply:
Good call - I am just pulling the camera from robot not accessing any of these. Removed.